src/linker/Linker/MethodIL.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Mono.Collections.Generic;
+
+namespace Mono.Linker
+{
+	/// <summary>
+	/// This is a wrapper which should be used by anything accessing method's body - see LinkContext.GetMethodIL for more details.
+	/// Any accesses made throught this wrapper are considered "safe"/OK since the wrapper is only created
+	/// once all of the optimizations are applied.
+	/// </summary>
+	public readonly record struct MethodIL
+	{
+		public MethodIL () => throw new InvalidOperationException ();
```

Comment:
Why is it needed?

Reply:
it's not needed now - it was there when the struct wasn't readonly in some early version...